src/Makefile.am

```diff
@@ -51,18 +51,8 @@ lib_LTLIBRARIES += \
 	libnss_mdns6_minimal.la
 endif
 
-noinst_PROGRAMS = nss-test
-sources = util.c util.h
-
-if ENABLE_AVAHI
-sources += avahi.c avahi.h
-noinst_PROGRAMS += avahi-test
-endif
-
-if ENABLE_LEGACY
-sources += query.c query.h dns.c dns.h
-noinst_PROGRAMS += mdns-test
-endif
+noinst_PROGRAMS = nss-test avahi-test
```

Comment:
These aren't proper tests, they have hardcoded addresses in them. They could be made into tests but currently will just fail after a while.